src/python_testing/TC_MWOCTRL_2_2.py

```diff
@@ -0,0 +1,96 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+import chip.clusters as Clusters
+from chip.interaction_model import InteractionModelError, Status
+from matter_testing_support import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from mobly import asserts
+
+# This test requires several additional command line arguments
+# run with
+# --int-arg PIXIT_ENDPOINT:<endpoint>
+
+
+class TC_MWOCTRL_2_2(MatterBaseTest):
+
+    async def read_mwoctrl_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.MicrowaveOvenControl
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    def desc_TC_MWOCTRL_2_2(self) -> str:
+        return "[TC-MWOCTRL-2.2] Secondary functionality with DUT as Server"
+
+    def steps_TC_MWOCTRL_2_2(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done", is_commissioning=True),
+            TestStep(2, "Read the PowerSetting attribute"),
+            TestStep(3, "Send the SetCookingParameters command"),
+            TestStep(4, "Read and verify the PowerSetting attribute"),
+        ]
+        return steps
+
+    def pics_TC_MWOCTRL_2_2(self) -> list[str]:
+        pics = [
+            "MWOCTRL.S",
+        ]
+        return pics
+
+    @async_test_body
+    async def test_TC_MWOCTRL_2_2(self):
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        self.step(1)
+        attributes = Clusters.MicrowaveOvenControl.Attributes
+        feature_map = await self.read_mwoctrl_attribute_expect_success(endpoint=endpoint, attribute=attributes.FeatureMap)
+        features = Clusters.MicrowaveOvenControl.Bitmaps.Feature
+        commands = Clusters.Objects.MicrowaveOvenControl.Commands
+
+        feature_map = await self.read_mwoctrl_attribute_expect_success(endpoint=endpoint, attribute=attributes.FeatureMap)
+
+        if feature_map == features.kPowerAsNumber:
```

Comment:
Two things that comes to mind. First, this check will fail is any other features is enabled, is that the intention here? I would assume we should just check if the bit is set or not and not really care about any other of the features 🤔 
Second, I would suggest to do the "if not supported" then skip and then have the test continue in the regular flow if the check is valid. This seems okay for now, but I'm worried it can quickly end up in a series of nested if statements.

Of cause I'm extremely biased here, but for BOOLCFG I did this:
```
is_aud_feature_supported = feature_map & Clusters.BooleanStateConfiguration.Bitmaps.Feature.kAudible

self.step(3)
if not is_sprs_feature_supported:
    logging.info("AlarmSuppress feature not supported skipping test case")
    
    # Skipping all remainig steps
    for step in self.get_test_steps(self.current_test_info.name)[self.current_step_index:]:
        self.step(step.test_plan_number)
        logging.info("Test step skipped")

    return
else:
    logging.info("Test step skipped")
```

Reply:
The reason for this `if` is that this test case is only valid if only that one bit is set.  If other bits are set, this TC should be skipped.  The `if` only serves to gate the TC.  If any other feature bit or no feature bit is supported, the TC should be skipped altogether.

I get your point about using `if not` instead.  It would make it read better.

Assuming this is ok for now, I'd like to get this merged in the interest of having the script available for TE2 and restructure in a subsequent PR.

Reply:
I still think the check it self is broken here, from what I can see in the spec PWRNUM or WATTS are mutually exclusive, but PWRLMTS can also be supported when PWRNUM is supported, right?

So 0b101 and 0b001 are both valid cases to make this test case applicable? As I understand feature_map == features.kPowerAsNumber, it mandates that feature_map MUST be exactly 0b001, right? Hence why I suggest to verify that the bit is set instead of checking the exact value.

Reply:
Correct, PWRNUM and WATTS are mutually exclusive.  This test case is meant to test the functionality if PWRLMTS is not supported, hence the `==` and not the bit test.  Test case 2.3 tests the situation when PWRNUM and PWRLMTS are both supported.  Test case 1.1 checks for the validity of the combinations of the feature bits, so this check isn't really a check, it just skips the test if some other combination of feature bits is set.